src/uu/env/src/env.rs

```diff
@@ -323,10 +340,15 @@ impl EnvAppData {
     fn run_env(&mut self, original_args: impl uucore::Args) -> UResult<()> {
         let (original_args, matches) = self.parse_arguments(original_args)?;
 
-        let did_debug_printing_before = self.do_debug_printing; // could have been done already as part of the "-vS" string parsing
-        let do_debug_printing = self.do_debug_printing || matches.get_flag("debug");
-        if do_debug_printing && !did_debug_printing_before {
-            debug_print_args(&original_args);
+        self.do_debug_printing = self.do_debug_printing || (0 != matches.get_count("debug"));
+        self.do_input_debug_printing = self
+            .do_input_debug_printing
+            .or(Some(matches.get_count("debug") >= 2));
+        if let Some(value) = self.do_input_debug_printing {
+            if value {
+                debug_print_args(&original_args);
+                self.do_input_debug_printing = Some(false);
```

Comment:
CI says that this line isn't covered by the tests, and trying to get it to be executed runs into another problem:
```console
$ env -vvS "--debug --debug" echo true
split -S:  '--debug --debug'
 into:    '--debug'
     &    '--debug'
executing: echo
   arg[0]= 'echo'
   arg[1]= 'true'
true
$ cargo run env -vvS "--debug --debug" echo true
input args:
arg[0]: 'env'
arg[1]: '-vvS'
arg[2]: '--debug --debug'
arg[3]: 'echo'
arg[4]: 'true'
env: error: unexpected argument '--debug --debug' found

  tip: a similar argument exists: '--debug'

Usage: target/debug/coreutils env [OPTION]... [-] [NAME=VALUE]... [COMMAND [ARG]...]

For more information, try '--help'.
env: use -[v]S to pass options in shebang lines
[$? = 125]
```

Is that issue part of this PR, or is this a separate issue?

Reply:
I added tests for all debug printing cases.

Your other finding is a different topic. And I doubt that it reflects a real usecase. The GNU testsuite for string args (tests/env/env-S.pl) is quite extensive. But this case is not covered. Normally the -S and the string are connected. E.g. 
```shell
env "-vvS --debug --debug" echo true
```